app/views.py

```diff
@@ -28,23 +28,53 @@ def home():
     """Renders the home page."""
     if current_user.is_authenticated:
         return redirect(url_for('dashboard'))
-    form = LoginForm()
-    if form.validate_on_submit():
-        user = User.query.filter_by(email=form.email.data).first()
-        if user is None or not user.check_password(form.password.data):
+    loginForm = LoginForm(prefix='login')
+    codeForm = CodeForm(prefix='code')
+    if loginForm.validate_on_submit():
+        user = User.query.filter_by(email=loginForm.email.data).first()
+        if user is None or not user.check_password(loginForm.password.data):
             flash('Invalid username or password')
             return redirect(url_for('login'))
         login_user(user)
         return redirect(url_for('dashboard'))
-    return render_template('index.html', form=form)
+    return render_template('index.html', loginForm=loginForm, codeForm=codeForm)
+
+@app.route('/progressreport/<int:knewbie_id>', methods=['GET','POST'])
+def get_report():
+    """Renders the class page for students/educators."""
+    loginForm = CreateName(prefix='login')
+    codeForm = CreateName(prefix='code')
+    if codeForm.validate_on_submit():
+        code = User.query.filter_by(knewbie_id=codeForm.code.data).first()
+        if code is None:
+            flash('Invalid code')
+            return redirect(url_for('home'))
+    return redirect(url_for('progressreport'))
 
 @app.route('/dashboard')
 def dashboard():
     """Renders the dashboard page."""
+    codeForm = CodeForm(prefix='code')
     classForm = CreateName(prefix='class')
     quizForm = CreateName(prefix='quiz')
```

Comment:
```suggestion
    classForm = NameForm(prefix='class')
    quizForm = NameForm(prefix='quiz')
```
To reflect the changes in `forms.py`.